lambdas/search/index.py

```diff
@@ -50,23 +51,51 @@ def lambda_handler(request):
     """
 
     action = request.args.get('action')
-    indexes = request.args.get('index')
-    terminate_after = os.getenv('MAX_DOCUMENTS_PER_SHARD')
-
-    if action == 'search':
+    user_body = request.args.get('body', {})
+    user_fields = request.args.get('fields', [])
+    user_indexes = request.args.get('index', "")
+    user_size = request.args.get('size', DEFAULT_SIZE)
+    user_source = request.args.get('_source', [])
+    terminate_after = None  # see if we can skip os.getenv('MAX_DOCUMENTS_PER_SHARD')
+
+    if action == 'packages':
+        query = request.args.get('query', '')
+        body = user_body or {
+            "query": {
+                "query_string": {
+                    "analyze_wildcard": True,
+                    "lenient": True,
+                    "query": query,
+                    # see enterprise/**/bucket.py for mappings
+                    "fields": user_fields or [
+                        # package
+                        'comment', 'handle', 'handle_text^2', 'metadata_string', 'tags'
+                    ]
+                }
+            }
+        }
+        _source = user_source
+        size = user_size
+    elif action == 'search':
         query = request.args.get('query', '')
         body = {
             "query": {
-                "simple_query_string": {
+                "query_string": {
```

Comment:
Changing from simple_query_string to query_string will cause Elastic to throw an exception if users submit bad queries. https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-simple-query-string-query.html The fault-tolerant aspect of simple_query_string is probably a better user experience.

Reply:
Backlogged for better search errors